src/main/java/seedu/commands/FilterCommand.java

```diff
@@ -0,0 +1,56 @@
+package seedu.commands;
+import seedu.duke.Internship;
+import seedu.duke.InternshipList;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FilterCommand implements Command {
+    private final InternshipList internships;
+    private InternshipList filteredInternships;
+
+    public FilterCommand(InternshipList internshipList) {
+        this.internships = internshipList;
+        this.filteredInternships = new InternshipList();
+    }
+
+    @Override
+    public void execute(String[] args) {
+        if (args.length > 1) {
+            String flag = args[0];        // The flag (e.g., -role, -company)
+            String searchTerm = args[1];  // What you are searching for
+
+            // Map flags to getter methods using lambdas
+            Map<String, InternshipFieldGetter> fieldGetters = new HashMap<>();
```

Comment:
This is completely optional, but i think defining <code>fieldGetters</code> can be done outside of the <code>execute</code> function, should make the function more readable.